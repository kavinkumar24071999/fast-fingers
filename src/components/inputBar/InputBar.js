import React, {useState} from 'react';

const SPACE = ' ';
const EMPTY = '';

const InputBar = () => {
  const [word, setWord] = useState(EMPTY);

  const handleChange = (word) => {
    if (word.includes(SPACE)) {
      setWord(EMPTY);
    } else {
      setWord(word);
    }
  };

  return (
    <div className="input-bar">
      <input value={word || EMPTY} onChange={(event) => handleChange(event.target.value)} />
    </div>
  );
};

export default InputBar;

