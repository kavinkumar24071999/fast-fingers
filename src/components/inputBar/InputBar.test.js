import Enzyme, {shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import InputBar from './InputBar';

Enzyme.configure({adapter: new Adapter()});

describe('Input bar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<InputBar />);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have input field', () => {
    const wrapper = shallow(<InputBar />);

    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('should change input on change of value', () => {
    const wrapper = shallow(<InputBar />);

    const event = {target: {value: 'inp'}};
    wrapper.find('input').simulate('change', event);

    expect(wrapper.find('input').prop('value')).toEqual("inp");
  });

  it('should clear input when the value contains space', () => {
    const wrapper = shallow(<InputBar />);

    const event = {target: {value: 'inp '}};
    wrapper.find('input').simulate('change', event);

    expect(wrapper.find('input').prop('value')).toEqual("");
  });
});
