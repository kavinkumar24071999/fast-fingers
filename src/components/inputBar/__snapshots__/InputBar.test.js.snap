// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input bar should match snapshot 1`] = `"<div class=\\"input-bar\\"><input value=\\"\\"/></div>"`;
